SD-949 rename r_m_a to its fullname return_merchandise_authorization

KacperMekarski opened the pull request:
Reported here: https://github.com/spree/spree/issues/10456

The problem was that class RMARequired in classic reload was converted to rma_required to find a matching file, while zeitwerk required file named r_m_a_required. In my opinion best solution is to use fullname of rma (return_merchandise_authorization) which solves the problem of converting class to snake case in both cases.

Update: as brunovelasco mentioned, adding require_relative 'r_m_a_required' to core/app/models/spree/return_item/eligibility_validator/default.rb did not solve the problem.

I don't have any good idea how to test that using RSpec.

Commit:
SD-949 rename r_m_a to its fullname return_merchandise_authorization

Commit:
SD-949 rename r_m_a to its fullname return_merchandise_authorization

damianlegawiec reviewed (commented):
@KacperMekarski we need to fix this in a way we're not going to change the class name - this will break a lot of current spree instances, decorators and so on

damianlegawiec reviewed (changes requested):
@KacperMekarski 

This can be achieved by creating a `Zeitwerk::Inflector` initializer, eg. https://github.com/solidusio/solidus/blob/466ac8e4b3217dca86bc2194b3dcff56111c3e0c/core/config/initializers/inflections.rb#L3

hoshinotsuyoshi committed:
Set `user_id` to new address

Because Spree::Address#clone does not copy `user_id` by default now.

https://github.com/spree/spree/issues/10569

Commit:
Merge branch 'master' into SD-949-uninitialized-constant-rma-required

Commit:
Fix 'ensure_order_token' accepting empty string as valid input

damianlegawiec committed:
Merge pull request #10570 from route06/fix_user_address_updating

Fix #10569 : Set `user_id` to new address in `PATCH /addresses/:id`

KacperMekarski committed:
Sd 962 unassociated addresses (#10543)

* SD-962 add tests

* SD-962 add checkout address tests, user is assigned to address without shipment and delivery mode

* SD-962 refactor update_or_create_address method and tests

* SD-962 add tests when address attributes are invalid

* SD-962 rubocop adjustments

* SD-962 small refactor in update_or_create_address method

* SD-962 remove commented code

* SD-962 saving user address depends on save_user_address param

* SD-962 refactor normalize_address method to pass through rubocop validations

* SD-962 fix normalize_address method

* SD-962 refactor update_or_create_address method

* SD-962 refactor normalize_address method

* SD-962 refactor normalize_address method

* Change default address scope based on current user in update_or_create_address, add checking for current user in normalize_address

damianlegawiec committed:
Merge pull request #10573 from upsidelab/fix/order_status_accepting_blank_order_token

Fix security issue in order status endpoint

Commit:
SD-949 add RMA as acronym

Commit:
SD-949 add RMA as acronym

Commit:
SD-949 add empty line

Commit:
SD-949 add empty line

Commit:
SD-949 add single quoted string

Commit:
SD-949 add single quoted string

KacperMekarski commented:
@damianlegawiec done.

Works for both classic reload & zeitwerk mode

KacperMekarski commented on core/app/models/spree/return_item/eligibility_validator/default.rb:
```diff
@@ -1,5 +1,3 @@
-require_relative 'r_m_a_required'
```
I noticed that `require_relative` does the trick for file named `rma_required` (then adding acronym is not neccessary)

Commit:
Merge branch 'SD-949-uninitialized-constant-rma-required' of github.com:spark-solutions/spree into SD-949-uninitialized-constant-rma-required

Commit:
SD-949 add require_relative rma_required instead of acronym

KacperMekarski commented:
@damianlegawiec I rebased from `master`, but tests failed due to error from db migrations:
> Caused by:
NameError: uninitialized constant AddStockLocationToRMA

caused probably by added acronym.
Therefore I used second solution which I mentioned here: https://github.com/spree/spree/pull/10527#discussion_r520449256 which works fine.

KacperMekarski commented:
Created new PR: https://github.com/spree/spree/pull/10574
